### ISSUE TITLE
pwa-node race condition, skipping breakpoints

### ISSUE THREAD

**Issue by makingglitches** [microsoft/vscode-js-debug#718]
**Describe the bug**

In the 'newest' version of vscode, pwa-node attach and launch alone with an explicit port defined.

```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Attach",
            "port":51061,
            "request": "attach",
            "skipFiles": [
                "<node_internals>/**"
            ],
            "type": "pwa-node"
        },
        
        {
            "name": "Launch Program",
            "program": "${workspaceFolder}/parseuszips.js",
            "request": "launch",
            "skipFiles": [
                "<node_internals>/**"
            ],
            "type": "pwa-node"
        },   
    ]
}
```

I have no idea why the debugger is choosing that port, but it does not connect otherwise to the debugging session.
Anyway the major issue is in the title, the breakpoints are not hit.
And this place is filled with very strange people we dont like.

Anyway, earlier breakpoints are not being hit, the debugging process is not waiting for the ide to attach,
Later breakpoints however are reached,

**To Reproduce**
Create pwa-node type for attach and launch in launch,json file under .vscode

Set breakpoints throughout the file 

Run the app.

**Log File**

[vscode-debugadapter-0.json.gz](https://github.com/microsoft/vscode-js-debug/files/5111874/vscode-debugadapter-0.json.gz)

**VS Code Version:** 1.48.1

**Comment by makingglitches** [comment]
Yes indeed after OS restart as well the issue remains, just thought to add that.
Btw previously when encountering this crap bug, did actually attempt to trace through the source of vscode, quite correct the code is more complex than a few hours of familiarization would allow in my presently kind of exhausted state. not to mention distracted by ugly monster people :-)

[vscode-debugadapter-0.json.gz](https://github.com/microsoft/vscode-js-debug/files/5136475/vscode-debugadapter-0.json.gz)

**Comment by makingglitches** [comment]
wouldn't life be grand if people just left shit stand, not like their constant dodging crap is doing them or anyone else any favors.

**Comment by connor4312** [comment]
Okay, thanks for the log. One thing to try: can you set the property `outFiles: []` in your launch.json?

**Comment by makingglitches** [comment]
no typescript based javascript is being run. the program is generating adapter classes for a json dataset in typescript, just happen to have tscconfig included in project for later testing.

**Comment by connor4312** [comment]
Okay, that's fine. What happens when setting outFiles to an empty array?

**Comment by makingglitches** [comment]
why would that matter if there is no generated javascript code being included and run ?

**Comment by makingglitches** [comment]
downgrading from the willfully bugged version back one minor version number without the cons worked. 
let us know when this is figured out.
hopefully by qualified individuals.

**Comment by connor4312** [comment]
Thanks for the info that downgrading works. Were you able to try setting outfiles to an empty array as described?

**Comment by connor4312** [comment]
Actually, I was able to repro this in Node 10.1.0, which is what you're using. This is a known bug in Node, which I thought was fixed by 10.0.0 but is apparently in a minor point release.

You can safely update to a more recent Node 10.x version (such as 10.22.0), or update to Node 14 for additional features and performance.
